src/tests/Register.test.js

```diff
@@ -106,4 +107,70 @@ it('blocks the user to register without password confirmation', async () => {
 
     expect(screen.queryByLabelText("response")).not.toBeInTheDocument();
     
-})
\ No newline at end of file
+})
+
+it('checks empty field warning', async () => {
+
+    render(<Register />)
+
+
+    fireEvent.change(screen.getByLabelText("Name :", { selector: "input" }), {
+        target: { value: '' },
+    })
+
+
+    fireEvent.change(screen.getByLabelText("Username :", { selector: "input" }), {
+        target: { value: '' },
+    })
+
+    fireEvent.change(screen.getByLabelText("Email :", { selector: "input" }), {
+        target: { value: '' },
+    })
+
+    fireEvent.change(screen.getByLabelText("Password :", { selector: "input" }), {
+        target: { value: '' },
+    })
+
+    act(() => {
+        fireEvent.click(screen.getByRole('button', { name: "Sign Up" }), {
+            target: { value: 'true' },
+        })
+    });
+    expect(screen.getByLabelText("Name :", { selector: "input" })).toBeRequired();
+    expect(screen.getByLabelText("Username :", { selector: "input" })).toBeRequired();
+    expect(screen.getByLabelText("Email :", { selector: "input" })).toBeRequired();
+    expect(screen.getByLabelText("Password :", { selector: "input" })).toBeRequired();
+})
+
+it('checks validation message warning', async () => {
+
+    render(<Register />)
+
+
+    fireEvent.change(screen.getByLabelText("Name :", { selector: "input" }), {
+        target: { value: '%' },
+    })
+
+
+    fireEvent.change(screen.getByLabelText("Username :", { selector: "input" }), {
+        target: { value: '&' },
+    })
+
+    fireEvent.change(screen.getByLabelText("Email :", { selector: "input" }), {
+        target: { value: 'mmo' },
+    })
+
+    fireEvent.change(screen.getByLabelText("Password :", { selector: "input" }), {
+        target: { value: '1' },
+    })
+
+    act(() => {
+        fireEvent.click(screen.getByRole('button', { name: "Sign Up" }), {
+            target: { value: 'true' },
+        })
+    });
+    expect(screen.getByLabelText("Name :", { selector: "span" })).toHaveTextContent("Must be between 2-30 characters long. Can only contain alphabets, whitespace and dash '-'");
```

Comment:
From line 172-175, you are targetting the error message inside the spans, not the input fields. So, use aria-label attribute on each span like how it was used on responseMessage in Register.jsx line 248.

<img width="1364" alt="Screen Shot 2020-09-08 at 6 35 38 pm" src="https://user-images.githubusercontent.com/29667122/92453144-42a22380-f202-11ea-8f39-0363a0a11439.png">

And you can use it inside test likee how it was used in first test `allows the user to register successfully` line 69.
<img width="827" alt="Screen Shot 2020-09-08 at 6 39 21 pm" src="https://user-images.githubusercontent.com/29667122/92453810-0fac5f80-f203-11ea-96ec-cae1dbc2085f.png">

Reply:
@mtreacy002 can I use the following code for validation message warning all the tests are passing with this code
![image](https://user-images.githubusercontent.com/66299533/92553419-80de2800-f280-11ea-9791-d2cde5260abe.png)
cuz I did every change as u requested me to but, the password test is still failing, the other one seems to work fine.

Reply:
If it works for you, then go ahead with it 👍

Reply:
Don't forget to push the latest change.

Reply:
> Don't forget to push the latest change.

Yes sure